### DISCUSSION TITLE
How can we add rounded corners to images?

### DISCUSSION THREAD

**Discussion by Wind-Explorer** [AvaloniaUI/Avalonia#9980] [Q&A] [answered]
In Avalonia, we can use the `CornerRadius` markup tag to round several elements, including `Border`, `Button`, `ListBox` and more. Now I have a UI layout in which I think adding the same rounded corners to `Image` would look nice. However I see that `CornerRadius` is not an option under `<Image/>`.

![Image showing the inability to use CornerRadius markup tag for image.](https://cdn.discordapp.com/attachments/918276992812605541/1064075112606859315/2023-01-15_2.54.08.png)

Even after wrapping the `Image` inside a `Border` with `CornerRadius`, the `Image` still appears to have no rounded corners.

![Image showing the lack of effect on the corners of an image wrapped in a border with cornerradius set to 10.](https://cdn.discordapp.com/attachments/918276992812605541/1064075565688168468/2023-01-15_2.56.21.png)

There are somehow no relevant information I can find regarding this. Am I not doing it the right way, or is this not possible in the current version of Avalonia?

**Comment by maxkatz6** [comment, marked as answer]
Set Border ClipToBounds to true, so it can clip its children.

**Reply by Wind-Explorer** [reply]
Thank you for the solution, works great! Though it would be nice if one day `CornerRadius` can be available for `Image` in Avalonia.

**Reply by ghost** [reply]
@Wind-Explorer Yeah that would be very helpful!

**Reply by Dawson924** [reply]
My problem solved! thank you for the solution*^____^*

**Comment by stany24** [comment]
You can easily add the CornerRadius property to image like this:

```
using Avalonia;
using Avalonia.Controls;
using Avalonia.Media;

namespace DesktopClient.Controls;

public class RoundedImage:Image
{
    public static readonly AttachedProperty<double> CornerRadiusProperty =
        AvaloniaProperty.RegisterAttached<RoundedImage, double>(
            "CornerRadius", typeof(RoundedImage), 5);
    
    public static void SetCornerRadius(AvaloniaObject element, double parameter)
    {
        element.SetValue(CornerRadiusProperty, parameter);
    }

    public static double GetCornerRadius(AvaloniaObject element)
    {
        return element.GetValue(CornerRadiusProperty);
    }
    
    protected override Size MeasureOverride(Size availableSize)
    {
        IImage? source = Source;
        Size result = new();

        if (source != null)
        {
            result = Stretch.CalculateSize(availableSize, source.Size, StretchDirection);
        }
        Clip = new RectangleGeometry(new Rect(0, 0, result.Width, result.Height), GetCornerRadius(this),GetCornerRadius(this));
        return result;
    }
}
```
